src/Atmos/Model/moisture.jl

```diff
@@ -79,11 +79,9 @@ soundspeed(atmos::AtmosModel, ::MoistureModel, state::Vars, aux::Vars) =
     state::Vars,
     aux::Vars,
 )
-    phase = thermo_state(atmos, state, aux)
-    R_m = gas_constant_air(phase)
-    T = air_temperature(phase)
-    e_tot = state.ρe * (1 / state.ρ)
-    return e_tot + R_m * T
+    ts = thermo_state(atmos, state, aux)
+    e_tot = total_specific_energy(state)
+    return total_specific_enthalpy(e_tot, ts)
```

Comment:
Do we need the `total_specific_enthalpy` here, in `moisture.jl`? Or can we move the whole thing to `Thermodynamics`, so it can be near its close cousins, the other thermodynamic energy functions?

Reply:
I think it's fine to have both. This is kind of a convenience wrapper and some methods already depend on this interface. I imagine that we'll have wrapper methods for most/all thermodynamic functions somewhere in `Atmos/`, which utilize `atmos`/`state`/`aux` so that all thermo methods can be called with these 3 arguments.

Reply:
This was actually only used in one place, so I've removed it and called the thermo method directly.